GX-TimeFrequency/src/minimax_grids.F90

```diff
@@ -166,29 +141,30 @@ subroutine gx_minimax_grid(num_points, e_min, e_max, &
 
     ! Compute the actual weights used for the inhomogeneous cosine/ FT and check whether
     ! the two matrices for the forward/backward transform are the inverse of each other.
-    if(.not.my_bare_cossine_weights) then
-      do j_point = 1, num_points
-         do i_point = 1, num_points
-            cosft_wt(j_point, i_point) = cosft_wt(j_point, i_point)*cos(tau_points(i_point)*omega_points(j_point))
-            cosft_tw(i_point, j_point) = cosft_tw(i_point, j_point)*cos(tau_points(i_point)*omega_points(j_point))
-            sinft_wt(j_point, i_point) = sinft_wt(j_point, i_point)*sin(tau_points(i_point)*omega_points(j_point))
-         end do
-      end do
+    if(.not.my_bare_cos_sin_weights) then
```

Comment:
This is wrong now. The sin_ft includes now also sin(wt) factor if bare_cosine_weights = false, which is incorrect.. Same for cosft and cos_wt. Yes, maybe the code could be a little more elegant here, but it is really easy to introduce bugs (which I had because of a similar issue) . Would require testing with cp2k etc. I strongly advocate not to touch it for now. However, we can think about adding a test which also captures the flag. I will also add a regression test in cp2k (once the library is released) which runs GreenX test. This should also make code beauty changes safer